trieste/models/gpflow/models.py

```diff
@@ -507,13 +496,291 @@ def conditional_predict_y(
+        inducing variables, i.e. `q_mu` and `q_sqrt` accordingly. The exact form of this update
+        depends if we are using whitened representations of the inducing variables.
+        See :meth:`_whiten_points` for details.
+
+        :param new_inducing_points: The desired values for the new inducing points.
+        :raise NotImplementedError: If we try to update the inducing variables of a model
+            that has more than one set of inducing points.
+        """
+
+        if isinstance(new_inducing_points, list):
+            raise NotImplementedError(
+                f"""
+                We do not currently support updating models with multiple sets of
+                inducing points however received; {new_inducing_points}
+                """
+            )
+
+        old_inducing_points, _, _, _ = self.get_inducing_variables()
+        tf.assert_equal(
+            tf.shape(old_inducing_points), tf.shape(new_inducing_points)
+        )  # number of inducing points must not change
+
+        if isinstance(self.model.inducing_variable, SharedIndependentInducingVariables):
+            # gpflow says inducing_variable might be a ndarray; it won't
+            cast(TensorType, self.model.inducing_variable.inducing_variable).Z.assign(
+                new_inducing_points
+            )  # [M, D]
+        else:
+            self.model.inducing_variable.Z.assign(new_inducing_points)  # [M, D]
+
+    def get_inducing_variables(
+        self,
+    ) -> Tuple[Union[TensorType, list[TensorType]], TensorType, TensorType, bool]:
+        """
+        Return the model's inducing variables.
+
+        :return: The inducing points (i.e. locations of the inducing variables), as a Tensor or a
+            list of Tensors (when the model has multiple inducing points); A tensor containing the
+            variational mean q_mu; a tensor containing the Cholesky decomposition of the variational
+            covariance q_sqrt; and a bool denoting if we are using whitened or
+            non-whitened representations.
+        """
+        inducing_variable = self.model.inducing_variable
+
+        if isinstance(inducing_variable, SharedIndependentInducingVariables):
+            # gpflow says inducing_variable might be a ndarray; it won't
+            inducing_points = cast(TensorType, inducing_variable.inducing_variable).Z  # [M, D]
+        elif isinstance(inducing_variable, SeparateIndependentInducingVariables):
+            inducing_points = [
+                cast(TensorType, inducing_variable).Z
+                for inducing_variable in inducing_variable.inducing_variables
+            ]  # list of L [M, D] tensors
+        else:
+            inducing_points = inducing_variable.Z  # [M, D]
+
+        q_mu, q_var = self.model.compute_qu()
+        q_sqrt = tf.linalg.cholesky(q_var)
+        whiten = False  # GPflow's SGPR model does not use the whitened representation
```

Comment:
since I'm currently reusing `_covariance_between_points_for_variational_models` function for computing the convariance I'm computing the `q` objects here, using the `compute_qu` - thanks @sebastianober for the suggestion!